receiver/awscloudwatchmetricsreceiver/config_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscloudwatchmetricsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscloudwatchmetricsreceiver"
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidate(t *testing.T) {
+	cases := []struct {
+		name        string
+		config      Config
+		expectedErr error
+	}{
+		{
+			name: "Valid config",
+			config: Config{
+				Region:       "eu-west-1",
+				PollInterval: time.Minute * 5,
+				Metrics: &MetricsConfig{
+					Names: []*NamedConfig{{
+						Namespace:      "AWS/EC2",
+						MetricName:     "CPUUtilizaition",
+						Period:         time.Second * 60,
+						AwsAggregation: "Sum",
+						Dimensions: []MetricDimensionsConfig{{
+							Name:  "Test",
```

Comment:
Is there a real value that would have meaning to users? Tests are documentation as well, using real-world examples to help people understand the code under test.

Reply:
Agreed. I'll fix this

Reply:
I'll fix the remaining issues and rebase tomorrow